### PR TITLE
Included Support of OPA Gatekeeper

### LINKED ISSUE #314: Add Support of OPA Gatekeeper
Arakde has no support of [Open Policy Agent(OPA) GateKeeper](https://github.com/open-policy-agent/gatekeeper)
## Expected Behaviour

User should be able to install OPA from arkade
## Current Behaviour

There is no installation support of OPA
## Possible Solution

Add installation support for OPA

### PR DESCRIPTION
**Pull request by nitishkumar71**
Signed-off-by: Nitishkumar Singh <nitishkumarsingh71@gmail.com>

Users can now install OPA gatekeeper using arkade
## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
Closes #314 

## How Has This Been Tested?

1. Install Gatekeeper
```arkade install opa-gatekeeper```
2. Create OPA constraint template
```
kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper/master/demo/basic/templates/k8srequiredlabels_template.yaml
```
3. Create rule based on above created constraint template
```
kubectl apply -f https://raw.githubusercontent.com/open-policy-agent/gatekeeper/master/demo/basic/constraints/all_ns_must_have_gatekeeper.yaml
```
The above steps ensure that gatekeeper is installed successfully.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

- [ ] I have tested this on arm, or have added code to prevent deployment

### CODE REVIEW

**Comment by Waterdrips** [comment]
@nitishkumar71 Is the OPA Gatekeeper chart multi-arch? Has it been tested on arm64/armv7?

**Comment by nitishkumar71** [comment]
> @nitishkumar71 Is the OPA Gatekeeper chart multi-arch? Has it been tested on arm64/armv7?

I haven't tested in arm64/armv7 personally. but they do support these arch. Please look at their [dockerhub](https://hub.docker.com/r/openpolicyagent/gatekeeper/tags?page=1&ordering=last_updated) Images arch.

As well as their [makefile](https://github.com/open-policy-agent/gatekeeper/blob/db5a58054ceb07a1a63d951e2bacd9b21aad9076/Makefile#L239) include build release for these platforms

**Review by Waterdrips — APPROVED** [review]

**Comment by alexellis** [comment]
I would like to see OPA release a configurable chart, and that would make a great app. However these two static manifests don't look like a very generic example, it appears to just install a hard-coded sample. 

Can you help me understand what the app does @nitishkumar71 ?

**Comment by nitishkumar71** [comment]
> I would like to see OPA release a configurable chart, and that would make a great app. However these two static manifests don't look like a very generic example, it appears to just install a hard-coded sample.
> 
> Can you help me understand what the app does @nitishkumar71 ?

Gatekeeper allows to create templates for policies, which can be used to create multiple policies with different parameters. It helps in creation of generic rule across organization where different teams can pass different parameters as per their requirement.

Their are some standard [templates](https://github.com/open-policy-agent/gatekeeper-library) being managed by community, you can have a look on them.

**Comment by Waterdrips** [comment]
> I would like to see OPA release a configurable chart, and that would make a great app. However these two static manifests don't look like a very generic example, it appears to just install a hard-coded sample.
> 
> Can you help me understand what the app does @nitishkumar71 ?

Not sure I understand @alexellis - This is using the OPA helm chart.

**Comment by alexellis** [comment]
It did not seem to be when I reviewed it. Can you check if this helm chart is current and maintained?

**Comment by Waterdrips** [comment]
This is the official OPA Gatekeeper chart - Same one I use in other projects.

File: cmd/apps/opa_gatekeeper.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+		updateRepo, _ := command.Flags().GetBool("update-repo")
+		persistence, _ := command.Flags().GetBool("persistence")
+		customFlags, _ := command.Flags().GetStringArray("set")
+
+		fmt.Printf("Using kubeconfig: %s\n", kubeConfigPath)
+		arch := k8s.GetNodeArchitecture()
+		fmt.Printf("Node architecture: %q\n", arch)
+
+		overrides := map[string]string{}
+		overrides["persistence.enabled"] = strings.ToLower(strconv.FormatBool(persistence))
```
Comment (Waterdrips) [inline cmd/apps/opa_gatekeeper.go:75]: I think this is a copy-paste error - I can't see persistence as a chart option. (So we dont need the arkade flag or this override)

**Review by Waterdrips — CHANGES_REQUESTED** [review]

File: cmd/apps/opa_gatekeeper.go (hunk 1 above)
Comment (nitishkumar71) [reply cmd/apps/opa_gatekeeper.go]: Resolved

**Comment by alexellis** [comment]
Sounds good to me. Let me know if you are happy with the changes you requested and then let's get this merged.

**Review by Waterdrips — APPROVED** [review]

**Pull request merged**